### ISSUE TITLE
Generate field identifiers for survey schema dynamically

### ISSUE THREAD

**Issue by empicano** [mse-survey-tool/mse-survey-backend#28]
In order to simplify the survey json configuration files we can dynamically generate field identifiers by numbering the fields level-wise instead of defining them ourselves. This would furthermore make the naming predictable, which is important for addressing the fields when during mailing.

**Comment by dostuffthatmatters** [comment]
What Strapi does is automatically generate the identifier from the label. You can still change the identifier if you want to:

![out](https://user-images.githubusercontent.com/29046316/83820473-1b046200-a6cd-11ea-9b8f-90fbe764e941.gif)

**Comment by empicano** [comment]
Good tip, we could also generate it from the title. My initial thought was to simply assign numbers:

```json
{
    "1": true,
    "2": {
        "1": true,
        "2": false
    }
}
```

To explain the problem, see **survey_2_format.py**: I need some kind of delimiter in order to reliably chain the identifiers when generating the change url. Dynamically generating them ourselves would make sure that they have a consistent format, and that some character e.g. `.` could be used as delimiter.

Why do you think an admin should be able to see and/or change the identifiers, when they are only used internally?

**Comment by dostuffthatmatters** [comment]
Yes, I see. Generating the identifier from the hierarchy is intuitive. 

Actually `-` (minus) would be a better delimiter.

We don't have to make the admin see and be able to change the identifier, but I would definitely include it in the config-json!

**Comment by empicano** [comment]
Onboard for the `-` delimiter.

Why do you think it's better to include it in the configuration? Including it in the configuration would automatically mean that it's editable, thus error prone (forgetting to change it when copy pasting, including a `-` in the identifier name). As don't think it would add value to the admin to be able to see/change it in the configuration files, thus we might as well hide it.

If it's because we would need to calculate it in the frontend and backend instead of reading it, I think the performance loss is negligible, if any (especially if we use a simple hierarchy counter), and for me the abstraction overweighs in benefit. What do you think?

**Comment by dostuffthatmatters** [comment]
Yes you are right, that this is a possible source of error.

I just thought that one should be able to see the ids used in communication from the config. However if we make the way of generating these intuitive it is fine!

The performance loss is negligible indeed.

Would you prefer to generate them like `1-2-2` or like `parentLabel-childLabel-grandChildLabel`?

**Comment by empicano** [comment]
How about we add an entry in the docs on it for advanced users (in the future when we have the basic docs on the configuration files :grin:)?

I'd generate them like `1-2-2` for now simply because its easier. If we see that that grows to be confusing for large surveys, changing it should be relatively easy. Works for you?

**Comment by dostuffthatmatters** [comment]
Yes, a doc entry would be helpful!

Yes sounds great! :)
